core/src/main/java/org/mapfish/print/map/tiled/wmts/WMTSLayer.java

```diff
@@ -185,4 +167,31 @@ public Double getLayerDpi() {
             return WMTSLayer.this.param.dpi;
         }
     }
+
+    /**
+     * Prepare the baseURL to make a request.
+     *
+     * @param commonURL Base URL
+     * @param matrixId matrixId
+     * @param row row
+     * @param col cold
+     * @param layerParam layerParam
+     */
+    public static URI createRestURI(final String commonURL, final String matrixId, final int row, final int col,
+                              final WMTSLayerParam layerParam) throws URISyntaxException {
+        String path = layerParam.baseURL;
+        if (layerParam.dimensions != null) {
+            for (int i = 0; i < layerParam.dimensions.length; i++) {
+                String dimension = layerParam.dimensions[i];
+                final String value = layerParam.dimensionParams.getString(dimension.toUpperCase());
+                path = path.replace("{" + dimension + "}", value);
+            }
+        }
+        path = path.replace("{TileMatrixSet}", layerParam.matrixSet);
```

Comment:
The `{style}` (without capital) is missing ...
See table 32 in the WMTS specifications.